src/js/express.js

```diff
@@ -64,98 +87,147 @@ gpii.express.registerComponentLineage = function (childComponent, expressCompone
     }
 };
 
-// Wire a child to its immediate descendants.
+/**
+ *
+ * @param entry {Object} - A single entry object, which is expected to contain a `nick` variable.
+ * @returns {String} - The contents of the `nick` variable.
+ *
+ * Static function for use with `Array.prototype.map`.  Extracts just the "nick" variable from the structure we construct in `getOrderedNicknames`.
+ *
+ */
+gpii.express.extractNicks = function (entry) {
+    return entry.nick;
+};
+
+/**
+ *
+ * @param nicknames {Object} - An unordered array of "nicknames" representing children of a given component.
+ * @param component {Object} - The component we will resolve the "nicknames" against.
+ * @returns {Array} - An array that contains all of the original "nicknames", but ordered by priority and namespace.
+ *
+ * Start with the nicknames each child component registered with us and combine it with the relevant priority data to
+ * produce an ordered array.
+ *
+ */
+gpii.express.getOrderedNicknames = function (nicknames, component) {
+    // Create an array that combines the "nicknames" in `that.childrenByParent[path]` with each component's hints about
+    // `priority` and `namespace`.
+    var nickNamesWithComponentPrioritiesAndNamespaces = [];
+    fluid.each(nicknames, function (childNickname) {
+        var childComponent = component[childNickname];
+        nickNamesWithComponentPrioritiesAndNamespaces.push({ nick: childNickname, namespace: childComponent.options.namespace, priority: childComponent.options.priority });
+    });
+
+    // Order the combined data by priority so that we have predictable control over which is loaded when, then map it
+    // back to a simple ordered array.
+    return gpii.express.orderByPriority(nickNamesWithComponentPrioritiesAndNamespaces).map(gpii.express.extractNicks);
```

Comment:
The function of "map" plus the custom function can be achieved by means of the framework utility fluid.getMembers - https://github.com/fluid-project/infusion/blob/master/src/framework/core/js/Fluid.js#L635